internal/tests/helper/testing_helper.go

```diff
@@ -435,7 +435,12 @@ func NewTestTcpServer(t *testing.T) *TestTcpServer {
 	ts.ln, err = net.Listen("tcp", ":0")
 	require.NoError(err)
 
-	go ts.run()
+	serverStarted := make(chan struct{})
+	go func() {
+		close(serverStarted)
+		ts.run()
+	}()
+	<-serverStarted
```

Comment:
Is this really useful? The serverStarted channel really only guarantees that the goroutine has started, which shouldn't give us much more confidence about the server starting. Can't we just try sending a TCP packet or something on the listener in a loop (e.g. `c := net.Dial(); c.Write(), c.Read()`) to see whether the server is ready?

Reply:
That is a great call out.